gemfirexd/client/src/main/java/io/snappydata/jdbc/TomcatConnectionPool.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright (c) 2017 SnappyData, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+
+package io.snappydata.jdbc;
+
+import com.sun.xml.internal.fastinfoset.stax.events.Util;
+import org.apache.tomcat.jdbc.pool.DataSource;
+import org.apache.tomcat.jdbc.pool.PoolProperties;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * A Connection Pool class internally uses the tomcat connection pooling
+ * library.
+ */
+class TomcatConnectionPool {
+
+    private static DataSource datasource;
+
+    static enum PoolProps {
+
+        DRIVER_NAME("pool-driverClassName", ClientDriver.class.getName()),
+        URL("pool-url", ""), // Compulsory field user must provide
```

Comment:
I don't think user should be allowed to override these two properties. They should be fixed for use with thrift ClientDriver URL.

Reply:
yes, we are not allowing user to override, even if user passes it in props, we are explicitly setting it to use thrift ClientDriver (in the ClientPoolDriver.connect() before calling Tomcat Pool)